core/runtime/pom.xml

```diff
@@ -103,7 +103,14 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-extension-processor</artifactId>
             <version>${project.version}</version>
-            <scope>provided</scope>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
```

Comment:
Without this the processor forced me to add JavaDoc for my test `@ConfigItem` which is very unexpected.

Reply:
The problem here that it was running for `src/test/java`, it's still running for `src/main/java`, see also: https://github.com/quarkusio/quarkus/blob/main/core/runtime/pom.xml#L233-L254

I think nobody noticed the side-effect for test yet, because I'm the first to add a test config item there.

Reply:
Disclaimer: `annotationProcessorPaths` is only set for the `compile` goal, so `test-compile` scans for all processors on the CP as per: https://maven.apache.org/plugins/maven-compiler-plugin/testCompile-mojo.html#annotationProcessorPaths